### DISCUSSION TITLE
Need help with --cli-input-json command

### DISCUSSION THREAD

**Discussion by adityasaurabh025** [aws/aws-cli#7585] [Q&A] [answered]
I am working on a Amazon Managed Blockchain(AMB) project, as a first step, I have to create blockchain member and a network. I have followed the below AMB documentation to install it on my ec2 instance, but not getting the expected output.

**Input:-**

aws managedblockchain create-network --cli-input-json "{"Name":"OurBlockchainNet","Description":"OurBlockchainNetDesc","Framework":"HYPERLEDGER_FABRIC","FrameworkVersion": "2.2","FrameworkConfiguration":{"Fabric":{"Edition": "STARTER"}},"VotingPolicy": {"ApprovalThresholdPolicy": {"ThresholdPercentage": 50, "ProposalDurationInHours": 24, "ThresholdComparator": "GREATER_THAN"}}, “MemberConfiguration”: {"Name":"org1", "Description":"Org1 first member of network", "FrameworkConfiguration":{"Fabric":n{"AdminUsername":"MyAdminUser","AdminPassword":"Password123"}}, "LogPublishingConfiguration": {"Fabric":{"CaLogs":{"Cloudwatch": {"Enabled": true}}}}}}"

**Current output:-**
To see help text, you can run:

  aws help
  aws <command> help
  aws <command> <subcommand> help

Unknown options: member, of, network, FrameworkConfiguration:{Fabric:n{AdminUsername:MyAdminUser,AdminPassword:Password123}}, LogPublishingConfiguration: {Fabric:{CaLogs:{Cloudwatch: {Enabled: true}}}}}}, first

**Expected output:-** 
{
    "NetworkId": "n-MWY63ZJZU5HGNCMBQER7IN6OIU",
    "MemberId": "m-K46ICRRXJRCGRNNS4ES4XUUS5A"
}

**AMB documentation:-** https://docs.aws.amazon.com/managed-blockchain/latest/hyperledger-fabric-dev/get-started-create-network.html

I am not sure what's the correct way to use --cli-input-json, because its throwing error in every possible way that I have tried.

Thanks!

**Comment by RyanFitzSimmonsAK** [comment, marked as answer]
Hi @adityasaurabh025, thanks for reaching out.  `--cli-input-json` seems to be expecting a file name as a parameter.  I was able to create a network using the command `aws managedblockchain create-network --cli-input-json file://test.json`, with my input file containing the code you pasted above, with a few typos fixed.  You have the wrong type of parentheses surrounding "MemberConfiguration", and you have a lowercase 'n' before your admin credentials.  

Please note that the example you copied from is using placeholder parameters.  You should definitely replace those, particularly the  admin username and password with something more secure.  Lastly, if you aren't using a region supported by AWS Managed Blockchain, you'll have to manually set a supported region using `--region`.

Hope that helps!  Let me know if you have any other questions.

**Comment by adityasaurabh025** [comment]
Hi Ryan, Thanks for your suggestion. I have fixed the issue now, took some help from chatGPT😁 and its resolved now. Thanks again for your help!
